src/main/kotlin/com/beust/nnk/NonLinearities.kt

```diff
@@ -0,0 +1,30 @@
+package com.beust.nnk
+
+data class NonLinearity(
+        val activate :((Float) -> Float),
+        val activateDerivative:((Float) -> Float)
+)
+
+object NonLinearities{
+    private val CommonNonLinearity= HashMap<String, NonLinearity> ()
```

Comment:
I was thinking an `enum` of the various activation strategies would be safer than a string?

Reply:
disclamer : I'm still a junior dev
I used this way once to allow the user to select from cli. Since, it became my "default".
I try to add security by the Exception and the UpperCase in the get method

Reply:
Autocomplation might work beter with an `enum` tho

Reply:
Using an `enum` doesn't prevent you from selecting the activation strategy from the command line (and actually, one of my other libraries, JCommander, supports enums natively :) ).

Not a huge deal, I'm happy to accept your merge request as it stands right now if you don't feel like making this change, I appreciate you taking the time to do this!